File: src_python/modeles/adresse.py
from dataclasses import dataclass
from province_ou_territoire import ProvinceOuTerritoire


@dataclass
class Adresse:
    numero_porte: str
    numero_rue: str
    nom_rue: str
    ville: str
    province: ProvinceOuTerritoire
    code_postal: str

    def __str__(self):
        return (
            f"Porte {self.numero_porte}, {self.numero_rue} rue {self.nom_rue}, "
            f"{self.ville}, {self.code_postal} {self.province.name}"
        )

File: TP2-eneonce-update/src/main/java/ca/uqam/mgl7460/tp1/types/modeles/Adresse.java
package ca.uqam.mgl7460.tp1.types.modeles;

/**
 * Quand on définit une "data class", c-à-d essentiellement des attributs, mais avec pas ou peu de méthode, on utilise
 * un <code>record</code> au lieu de <code>class</code>
 */
public record Adresse(String numeroPorte, String numeroRue, String nomRue, String ville, ProvinceOuTerritoire province, String codePostal) {
    public String toString() {
        return "Porte "+ numeroPorte+ ", "+ numeroRue+ " rue "+nomRue+ ", "+ville+ ", " + codePostal + " "+ province;
    }
}
